docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -1,219 +1,348 @@
-# Developing a new plug-in
+# Developing a new Zowe CLI plug-in
 Before you begin this tutorial, complete the [Extending an existing plug-in](cli-extending-a-plugin.md) tutorial.
 
 ## Overview
-This tutorial demonstrates how to create a brand new Zowe&trade; CLI plug-in that uses Zowe CLI Node.js programmatic APIs.
+The advantage of Zowe CLI and of the CLI approach in mainframe development is that it allows for combining different developer tools for new and interesting uses.
+
+This tutorial demonstrates how to create a brand new Zowe CLI plug-in that uses Node.js to create a client-side API.
+
+After following all the steps, you will have created a data set diff utility plug-in called **Files Util Plug-in**. This plug-in takes in any two data sets, or files, and returns a plain text output in the terminal showing how they differ. This tutorial will also show you how you can integrate your new plug-in with a third-party utility to make your output colorful and easier to read, as shown in the image at the [bottom of this page](../extend-cli/cli-developing-a-plugin#bringing-together-new-tools).
+
```

Comment:
[bottom of this page](../extend-cli/cli-developing-a-plugin#bringing-together-new-tools)
^ link doesn't work but i'm not sure if that will continue to happen outside of github? @t1m0thyj is this a problem?

Reply:
It works in VSC, so it'll work once the page goes live.